src/clojure/nrepl/bencode.clj

```diff
@@ -268,6 +268,16 @@
           :else (do (.unread input first-byte)
                     (read-netstring* input)))))
 
+(defn read-nrepl-message
+  "Sames as `read-bencode`, but ensure that the top-level value is a map as
+  expected by the nREPL protocol."
+  [^PushbackInputStream input]
+  (let [first-byte (read-byte input)]
+    (if (= first-byte d)
+      (read-map input)
+      (throw (ex-info (format "Top-level nREPL message must be a map.
```

Comment:
Does `top-level` makes sense in this context? I mean the message is a map, so I guess a simpler message will do.

Reply:
Fixed.